### PR TITLE
Provide missing information in JPlagResult and JPlagOptions objects

### LINKED ISSUE #248: Missing information in JPlagResult and JPlagOptions objects
The following information is missing in the JPlagResult/JPlagOptions object but it could be useful for the report:

## JPlagResult
- [x] **List of failed submissions.** Through the list size their amount can also be acquired.

## JPlagOption
- [x] **List of names of excluded files.** The object contains name of excluded file but it needs to be read in order to obtain the names. However the file is already being read one time when starting the JPlag comparison. I would suggest this read to be stored so we can avoid duplicated coded and multiple same file accesses at different places in the code.

### PR DESCRIPTION
**Pull request by tsaglam**
Resolves #248:
- Provide submission set via the JPlag result object
- ...

### CODE REVIEW

File: jplag/src/main/java/de/jplag/JPlagResult.java [hunk 1]
```diff
@@ -9,36 +9,15 @@
...
-     * Each entry represents the absolute frequency of matches whose similarity lies within the respective interval.
-     * <p>
-     * Intervals:
-     * <p>
-     * 0: [0% - 10%), 1: [10% - 20%), 2: [20% - 30%), ..., 9: [90% - 100%]
-     */
-    private int[] similarityDistribution = null;
+    private long durationInMillis;
+
+    private int[] similarityDistribution = null; // 10-element array representing the similarity distribution of the detected matches.
```
[the remark this answers is not in the document]
Comment (tsaglam) [reply jplag/src/main/java/de/jplag/JPlagResult.java:20]: Yes, but only if we either get rid of the default constructor or set all final fields to null in this constructor which is quite ugly.
The former would be probably nicer.

[the remark this answers is not in the document]
Comment (GabrielL) [reply jplag/src/main/java/de/jplag/JPlagResult.java:20]: From what I see, the default constructor does not seem to be used.

[the remark this answers is not in the document]
Comment (tsaglam) [reply jplag/src/main/java/de/jplag/JPlagResult.java:20]: You are right, it seems that we already got rid of all usages of the empty result.

File: jplag/src/main/java/de/jplag/options/JPlagOptions.java [hunk 2]
```diff
@@ -68,6 +70,11 @@
      * Name of the file that contains the names of files to exclude from comparison.
      */
     private String exclusionFileName;
+    
+    /**
+     * Names of the excluded files.
+     */
+    private Set<String> excludedFiles = Collections.emptySet();
```
Comment (GabrielL) [inline jplag/src/main/java/de/jplag/options/JPlagOptions.java:77]: Is it necessary to store more thing in the options ?

At first read, it looked like JPlagOptions is the command line parsing artefact, but it is written multiple times.

In this case, excluded files are already in `JPlag`, and directly passed into `SubmissionSetBuilder`, I don't see the need to add it.

Comment (tsaglam) [reply jplag/src/main/java/de/jplag/options/JPlagOptions.java:77]: See #192 and #274. We need the excluded files for the new report generation. Passing it through the core algorithms just to add it to the result object makes no sense. Passing it through the options is not ideal, but will do for now, as we already modify the options anyways.
If you have ideas regarding that, feel free to add it to #274.

**Pull request merged**